last_seen.py

```diff
@@ -0,0 +1,32 @@
+from brutal.core.plugin import BotPlugin, cmd, event
+from time import strftime, gmtime
+
+
+class LastSeen(BotPlugin):
+    def setup(self, *args, **kwargs):
+        self.seen = self.open_storage('last_seen')
+
+    @event
+    def seen_evt(self, event):
+        events = ['quit', 'part', 'kick', 'join', 'rename', 'topic', 'message']
+        if event.event_type in events:
+            nick = event.meta['nick']
+            time = gmtime()
+            evt_type = event.event_type
+            self.seen[nick] = {time: evt_type}
+
+    @cmd
+    def last_seen(self, event):
+        args = event.args
+        if len(args) < 1:
+            return "Name argument missing."
+        user = args[0]
+        seen = self.seen
+
+        if user not in seen:
+            return "This user was never here." + strftime("%H:%M %d.%m.%Y")
+
+        raw_time = seen[user].keys()[0]
+        evt = seen[user][raw_time]
+        time = strftime("%H:%M %d.%m.%Y", raw_time)
+        return "{0} was last seen: {1}({2})".format(user, time, evt)
```

Comment:
@adman suppose someone  hasn't  shown up for 20 days. I think x hours ago maes sense  only in the first 24 hours.